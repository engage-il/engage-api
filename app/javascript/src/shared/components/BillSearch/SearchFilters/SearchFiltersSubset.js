// @flow
import * as React from 'react'
import styled from 'react-emotion'
import { mixins, colors } from '@/styles'
import type { SearchParamsSubset } from '@/types'

type SubsetEntry = {
  label: string,
  value: SearchParamsSubset,
}

type Props = {
  entry: SubsetEntry,
  value: ?SearchParamsSubset,
  onChange: (event: SyntheticInputEvent) => void
}

export function SearchFiltersSubset ({
  entry,
  value,
  onChange
}: Props) {
  return (
    <Option>
      <input
        type='checkbox'
        name='subset'
        value={entry.value}
        onChange={onChange}
        checked={entry.value === value}
      />
      <span
        children={entry.label}
      />
    </Option>
  )
}

const Option = styled.label`
  ${mixins.flexRow};
  ${mixins.fonts.regularSlab};
  ${mixins.fontSizes.h5};

  align-items: center;
  color: ${colors.gray2};

  > input {
    margin-right: 10px;
  }
`
