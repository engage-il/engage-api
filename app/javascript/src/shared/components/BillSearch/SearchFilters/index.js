// @flow
export { SearchFilters } from './SearchFilters'
