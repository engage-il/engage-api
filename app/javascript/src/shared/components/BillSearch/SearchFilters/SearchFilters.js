// @flow
import * as React from 'react'
import styled from 'react-emotion'
import { SearchFiltersSubset } from './SearchFiltersSubset'
import { mixins, colors } from '@/styles'
import type { SearchParams } from '@/types'

type Props = {
  params: SearchParams,
  onChange: (SearchParams) => void
}

const SUBSET_ENTRIES = [{
  label: 'slips',
  value: 'SLIPS'
}, {
  label: 'house',
  value: 'LOWER'
}, {
  label: 'senate',
  value: 'UPPER'
}, {
  label: 'governor',
  value: 'GOVERNOR'
}]

export class SearchFilters extends React.Component<*, Props, *> {
  // events
  didChangeFilter = ({ target }: SyntheticInputEvent) => {
    const { params, onChange } = this.props

    onChange({
      ...params,
      [target.name]: target.checked ? target.value : null
    })
  }

  // lifecycle
  render () {
    const { params } = this.props

    return (
      <Filters>
        <Subsets>
          {SUBSET_ENTRIES.map((entry, i) => (
            <SearchFiltersSubset
              key={`filters-subset-${i}`}
              entry={entry}
              value={params.subset}
              onChange={this.didChangeFilter}
            />
          ))}
        </Subsets>
      </Filters>
    )
  }
}

const Filters = styled.div`
  ${mixins.flexRow};
  ${mixins.flexCenter};

  height: 120px;
  border-bottom: 1px solid ${colors.gray5};
`

const Subsets = styled.div`
  ${mixins.flexRow};
  align-items: center;

  > * + * {
    margin-left: 40px;
  }
`
