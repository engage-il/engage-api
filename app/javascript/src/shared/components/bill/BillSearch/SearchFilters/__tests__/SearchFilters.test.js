/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { defaultsDeep } from 'lodash'
import { SearchFilters } from '../SearchFilters'

// subject
let subject

const defaults = {
  params: {
    subset: 'test-subset',
    other: 'test-other'
  },
  onChange: jest.fn()
}

function loadSubject (props = {}) {
  subject = shallow(<SearchFilters {...defaultsDeep(props, defaults)} />)
}

// specs
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('shows the search filters', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})

describe('changing a filter', () => {
  it('sets the value when checked', () => {
    loadSubject()

    subject.instance().didChangeFilter({
      target: {
        name: 'subset',
        value: 'foo',
        checked: true
      }
    })

    expect(defaults.onChange).toHaveBeenCalledWith({
      ...defaults.params,
      subset: 'foo'
    })
  })

  it('clears the value when unchecked', () => {
    loadSubject()

    subject.instance().didChangeFilter({
      target: {
        name: 'subset',
        value: 'foo',
        checked: false
      }
    })

    expect(defaults.onChange).toHaveBeenCalledWith({
      ...defaults.params,
      subset: null
    })
  })
})
