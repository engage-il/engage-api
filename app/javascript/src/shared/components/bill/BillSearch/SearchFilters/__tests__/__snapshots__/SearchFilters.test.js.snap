// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render shows the search filters 1`] = `
<Filters>
  <Subsets>
    <SearchFiltersSubset
      entry={
        Object {
          "label": "slips",
          "value": "SLIPS",
        }
      }
      key="filters-subset-0"
      onChange={[Function]}
      value="test-subset"
    />
    <SearchFiltersSubset
      entry={
        Object {
          "label": "house",
          "value": "LOWER",
        }
      }
      key="filters-subset-1"
      onChange={[Function]}
      value="test-subset"
    />
    <SearchFiltersSubset
      entry={
        Object {
          "label": "senate",
          "value": "UPPER",
        }
      }
      key="filters-subset-2"
      onChange={[Function]}
      value="test-subset"
    />
    <SearchFiltersSubset
      entry={
        Object {
          "label": "governor",
          "value": "GOVERNOR",
        }
      }
      key="filters-subset-3"
      onChange={[Function]}
      value="test-subset"
    />
  </Subsets>
</Filters>
`;
